### PR TITLE
python311Packages.dm-haiku: 0.10.0 -> 0.11.0

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelog: https://github.com/google-deepmind/dm-haiku/releases/tag/v0.0.11

cc @ndl 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
The tests (`python311Packages.dm-haiku.tests`) are failing, but that was already the case before this update (exact same error):
```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
Sourcing pytest-check-hook
Using pytestCheckPhase
Removing setuptoolsCheckPhase
Running phase: unpackPhase
unpacking source archive /nix/store/n6gxl12zl30r0y7qs0ly76pzv3fa4mwy-python3.11-dm-haiku-0.0.11-testsout
source root is python3.11-dm-haiku-0.0.11-testsout
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file python3.11-dm-haiku-0.0.11-testsout/examples/vqvae_example.ipynb
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: pythonOutputDistPhase
Executing pythonOutputDistPhase
The build contains no ./dist directory.
If this project is not setuptools-based, pass

  format = "other";

to buildPythonApplication/buildPythonPackage or another appropriate value as described here:

  https://nixos.org/manual/nixpkgs/stable/#buildpythonpackage-function
```

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 279901` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.distrax</li>
    <li>python310Packages.distrax.dist</li>
    <li>python310Packages.dm-haiku</li>
    <li>python310Packages.dm-haiku.dist</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.distrax</li>
    <li>python311Packages.distrax.dist</li>
    <li>python311Packages.dm-haiku</li>
    <li>python311Packages.dm-haiku.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
  </ul>
</details>

**Comment by onny** [comment]
Really nice fix and update, LGTM

**Review by onny — APPROVED** [review]

File: pkgs/development/python-modules/bsuite/default.nix [hunk 1]
```diff
@@ -35,7 +36,14 @@ let bsuite = buildPythonPackage rec {
     hash = "sha256-ak9McvXl7Nz5toUaPaRaJek9lurxiQiIW209GnZEjX0=";
   };
 
-  buildInputs = [
+  patches = [
+    (fetchpatch {  # Convert np.int -> np.int32 since np.int is deprecated (https://github.com/google-deepmind/bsuite/pull/48)
+      url = "https://github.com/google-deepmind/bsuite/pull/48.patch";
```
Comment (NickCao) [inline pkgs/development/python-modules/bsuite/default.nix:41]: `pull/<id>.patch` urls are not immutable, please reference the commit directly.

Comment (GaetanLepage) [reply pkgs/development/python-modules/bsuite/default.nix]: Now I know !
Done.

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 279901` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.distrax</li>
    <li>python310Packages.distrax.dist</li>
    <li>python310Packages.dm-haiku</li>
    <li>python310Packages.dm-haiku.dist</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.distrax</li>
    <li>python311Packages.distrax.dist</li>
    <li>python311Packages.dm-haiku</li>
    <li>python311Packages.dm-haiku.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 279901` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.distrax</li>
    <li>python310Packages.distrax.dist</li>
    <li>python310Packages.dm-haiku</li>
    <li>python310Packages.dm-haiku.dist</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.distrax</li>
    <li>python311Packages.distrax.dist</li>
    <li>python311Packages.dm-haiku</li>
    <li>python311Packages.dm-haiku.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
  </ul>
</details>

**Pull request merged**